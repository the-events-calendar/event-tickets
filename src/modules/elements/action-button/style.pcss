.tribe-editor__action-button {

	&,
	/* extra class to override gutenberg styles */
	&.tribe-editor__button,
	&.tribe-editor__link {
		display: flex;
		align-items: center;
		font-family: 'Helvetica Neue', Helvetica, -apple-system, BlinkMacSystemFont, Roboto, Arial, sans-serif;
		font-size: 15px;
		line-height: 18px;
		letter-spacing: 0.38px;
		color: #545D66;
		text-decoration: none;
		transition: none;

		& > svg {
			width: 20px;
			fill: #545D66;

			path {
				fill: #545D66;
			}
		}

		&:hover,
		&:focus {
			color: #009fd4;

			& > svg {
				fill: #009fd4;

				path {
					fill: #009fd4;
				}
			}
		}

		&[disabled] {

			&,
			&:hover,
			&:focus {
				color: #8d949b;

				& > svg {
					fill: #8d949b;

					path {
						fill: #8d949b;
					}
				}
			}
		}

		.tribe-editor__action-dashboard & {

			&, &:hover, &:focus {
				box-shadow: none;
			}
		}
	}
}

.tribe-editor__action-button--icon-right {

	& > svg {
		margin-left: 6px;
		order: 2;
	}
}

.tribe-editor__action-button--icon-left {

	& > svg {
		margin-right: 6px;
	}
}
