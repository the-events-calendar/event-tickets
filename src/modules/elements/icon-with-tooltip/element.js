/**
 * External dependencies
 */
import React from 'react';
import PropTypes from 'prop-types';

/**
 * Internal dependencies
 */
import { Tooltip } from '@wordpress/components';

const IconWithTooltip = ( {
	description = '',
	icon,
	position = 'top right',
	propertyName,
} ) => {
	const text = (
		<div>
			{ propertyName }
			{ description && ': ' }
			{ description && <em>{description}</em> }
		</div>
	);

	return (
		<Tooltip
			text={ text }
			placement={ position }
		>
			<span>{ icon }</span>
		</Tooltip>
	);
};

IconWithTooltip.propTypes = {
	description: PropTypes.string,
	icon: PropTypes.node,
	position: PropTypes.oneOf( [
		'top left',
		'top center',
		'top right',
		'bottom left',
		'bottom center',
		'bottom right',
	] ),
	propertyName: PropTypes.string,
};

export default IconWithTooltip;
