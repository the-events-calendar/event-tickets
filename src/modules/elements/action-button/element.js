/**
 * External dependencies
 */
import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

/**
 * Internal dependencies
 */
import { Button, Link } from '@moderntribe/common/elements';
import './style.pcss';

export const positions = {
	right: 'right',
	left: 'left',
};

const components = {
	button: Button,
	link: Link,
};

const ActionButton = ( {
	asLink,
	children,
	className,
	disabled,
	href,
	icon,
	onClick,
	position,
	target,
	...props
} ) => {
	const containerClass = classNames(
		'tribe-editor__action-button',
		`tribe-editor__action-button--icon-${ position }`,
		className,
	);

	const Element = asLink && ! disabled ? components.link : components.button;

	const getProps = () => {
		const elemProps = { ...props };

		if ( asLink && ! disabled ) {
			elemProps.href = href;
			elemProps.target = target;
		} else {
			elemProps.disabled = disabled;
			elemProps.onClick = onClick;
		}

		return elemProps;
	};

	return (
		<Element
			className={ containerClass }
			{ ...getProps() }
		>
			{ icon }
			<span className="tribe-editor__action-button__label">{ children }</span>
		</Element>
	);
};

ActionButton.propTypes = {
	asLink: PropTypes.bool,
	children: PropTypes.node,
	className: PropTypes.string,
	disabled: PropTypes.bool,
	href: PropTypes.string,
	icon: PropTypes.node.isRequired,
	onClick: PropTypes.func,
	position: PropTypes.oneOf( Object.keys( positions ) ),
	target: PropTypes.string,
};

ActionButton.defaultProps = {
	asLink: false,
	position: positions.left,
};

export default ActionButton;
