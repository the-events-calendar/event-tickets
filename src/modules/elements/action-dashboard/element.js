/**
 * External dependencies
 */
import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { noop } from 'lodash';

/**
 * Internal dependencies
 */
import { Button } from '@moderntribe/common/elements';
import './style.pcss';

const ActionDashboard = ( {
	actions,
	cancelLabel,
	className,
	confirmLabel,
	isCancelDisabled,
	isConfirmDisabled,
	onCancelClick,
	onConfirmClick,
	showCancel,
	showConfirm,
} ) => {
	const actionsList = ( actions && !! actions.length ) && (
		<div className="tribe-editor__action-dashboard__group-left">
			{ actions.map( ( action, index ) => (
				<span
					key={ `action-${ index }` }
					className="tribe-editor__action-dashboard__action-wrapper"
				>
					{ action }
				</span>
			) ) }
		</div>
	);

	const cancelButton = showCancel && (
		<Button
			className="tribe-editor__action-dashboard__cancel-button"
			isDisabled={ isCancelDisabled }
			onClick={ onCancelClick }
		>
			{ cancelLabel }
		</Button>
	);

	const confirmButton = showConfirm && (
		<Button
			className="tribe-editor__action-dashboard__confirm-button tribe-editor__button--sm"
			isDisabled={ isConfirmDisabled }
			onClick={ onConfirmClick }
		>
			{ confirmLabel }
		</Button>
	);

	const groupRight = ( showCancel || showConfirm ) && (
		<div className="tribe-editor__action-dashboard__group-right">
			{ cancelButton }
			{ confirmButton }
		</div>
	);

	return (
		<section
			className={ classNames(
				'tribe-editor__action-dashboard',
				className,
			) }
		>
			{ actionsList }
			{ groupRight }
		</section>
	);
};

ActionDashboard.defaultProps = {
	showCancel: true,
	showConfirm: true,
	onCancelClick: noop,
	onConfirmClick: noop,
};

ActionDashboard.propTypes = {
	actions: PropTypes.arrayOf( PropTypes.node ),
	cancelLabel: PropTypes.string,
	className: PropTypes.string,
	confirmLabel: PropTypes.string,
	isCancelDisabled: PropTypes.bool,
	isConfirmDisabled: PropTypes.bool,
	onCancelClick: PropTypes.func,
	onConfirmClick: PropTypes.func,
	showCancel: PropTypes.bool,
	showConfirm: PropTypes.bool,
};

export default ActionDashboard;
