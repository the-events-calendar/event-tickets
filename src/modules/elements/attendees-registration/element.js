/**
 * External dependencies
 */
import React, { createRef, PureComponent } from 'react';
import PropTypes from 'prop-types';

/**
 * Wordpress dependencies
 */
import { Spinner } from '@wordpress/components';
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { LabelWithModal } from '@moderntribe/common/elements';
import './style.pcss';

class AttendeesRegistration extends PureComponent {
	static propTypes = {
		helperText: PropTypes.string.isRequired,
		iframeURL: PropTypes.string.isRequired,
		isDisabled: PropTypes.bool.isRequired,
		isModalOpen: PropTypes.bool.isRequired,
		label: PropTypes.string.isRequired,
		linkText: PropTypes.string.isRequired,
		modalTitle: PropTypes.string.isRequired,
		onClick: PropTypes.func.isRequired,
		onClose: PropTypes.func.isRequired,
		onIframeLoad: PropTypes.func.isRequired,
		showHelperText: PropTypes.bool.isRequired,
	};

	constructor( props ) {
		super( props );
		this.iframe = createRef();
	}

	render() {
		const {
			helperText,
			iframeURL,
			isDisabled,
			isModalOpen,
			label,
			linkText,
			modalTitle,
			onClick,
			onClose,
			onIframeLoad,
			showHelperText,
			...restProps
		} = this.props;

		const modalContent = (
			<div className="tribe-editor__attendee-registration__modal-content">
				<iframe
					className="tribe-editor__attendee-registration__modal-iframe"
					onLoad={ () => onIframeLoad( this.iframe.current ) }
					ref={ this.iframe }
					src={ iframeURL }
					title={ __( 'Attendee registration', 'event-tickets' ) }
				>
				</iframe>
				<div className="tribe-editor__attendee-registration__modal-overlay">
					<Spinner />
				</div>
			</div>
		);

		return (
			<div className="tribe-editor__attendee-registration">
				<LabelWithModal
					className="tribe-editor__attendee-registration__label-with-modal"
					isOpen={ isModalOpen }
					label={ label }
					modalButtonDisabled={ isDisabled }
					modalButtonLabel={ linkText }
					modalClassName="tribe-editor__attendee-registration__modal"
					modalContent={ modalContent }
					modalTitle={ modalTitle }
					onClick={ onClick }
					onClose={ onClose }
					{ ...restProps }
				/>
				{ showHelperText && (
					<span className="tribe-editor__attendee-registration__helper-text">
						{ helperText }
					</span>
				) }
			</div>
		);
	}
}

export default AttendeesRegistration;
