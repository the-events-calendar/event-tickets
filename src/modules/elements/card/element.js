/**
 * External dependencies
 */
import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

/**
 * Internal dependencies
 */
import './style.pcss';

const getHeaderElement = (header, description) => {
    if (!description) {
        return (<div className="tickets-heading tickets-row-line">{ header }</div>)
    }

    return (
        <div className="tickets-heading__wrapper tickets-row-line">
            <div className="tickets-heading tickets-heading__title">{ header }</div>
            <div className="tickets-heading__description">{ description }</div>
        </div>
    );
};

const Card = ({
                  className,
                  children,
                  header,
                  description
              }) => {
    return (
        <div className={ classNames('tribe-editor__card', className) }>
            { header && getHeaderElement(header, description) }
            { children }
        </div>
    );
};

Card.propTypes = {
    className: PropTypes.string,
    children: PropTypes.node,
    header: PropTypes.string,
    description: PropTypes.string,
};

export default Card;
