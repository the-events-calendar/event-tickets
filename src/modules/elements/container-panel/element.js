/**
 * External dependencies
 */
import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

/**
 * Internal dependencies
 */
import './style.pcss';

export const LAYOUT = {
	rsvp: 'rsvp',
	ticket: 'ticket',
};

const ContainerPanel = ( { className, content, header, layout } ) => {
	const headerAndContent = (
		<Fragment>
			<div className="tribe-editor__container-panel__header">{ header }</div>
			{ content && <div className="tribe-editor__container-panel__content">{ content }</div> }
		</Fragment>
	);

	const getHeaderAndContent = () =>
		layout === LAYOUT.ticket ? (
			headerAndContent
		) : (
			<div className="tribe-editor__container-panel__header-content-wrapper">{ headerAndContent }</div>
		);

	return (
		<div
			className={ classNames(
				'tribe-editor__container-panel',
				`tribe-editor__container-panel--${ layout }`,
				className
			) }
		>
			{ getHeaderAndContent() }
		</div>
	);
};

ContainerPanel.propTypes = {
	className: PropTypes.string,
	content: PropTypes.node,
	header: PropTypes.node,
	layout: PropTypes.oneOf( Object.keys( LAYOUT ) ).isRequired,
};

export default ContainerPanel;
