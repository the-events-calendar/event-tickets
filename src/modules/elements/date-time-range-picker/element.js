/**
 * External dependencies
 */
import React, { createRef, Component } from 'react';
import PropTypes from 'prop-types';
import { noop } from 'lodash';
import classNames from 'classnames';
import { formatDate, parseDate } from 'react-day-picker/moment';

/**
 * Internal dependencies
 */
import { DayPickerInput, TimePicker } from '@moderntribe/common/elements';
import {
	date,
	time,
} from '@moderntribe/common/utils';
import './style.pcss';

class DateTimeRangePicker extends Component {
	static defaultProps = {
		fromDateFormat: 'LL',
		onFromDateChange: noop,
		onToDateChange: noop,
		separatorDateTime: 'at',
		separatorTimeRange: 'to',
		toDateFormat: 'LL',
	};

	static propTypes = {
		className: PropTypes.string,
		fromDate: PropTypes.instanceOf( Date ),
		fromDateInput: PropTypes.string,
		fromDateDisabled: PropTypes.bool,
		fromDateFormat: PropTypes.string,
		fromTime: PropTypes.string,
		fromTimeDisabled: PropTypes.bool,
		onFromDateChange: PropTypes.func,
		onFromTimePickerBlur: PropTypes.func,
		onFromTimePickerChange: PropTypes.func,
		onFromTimePickerClick: PropTypes.func,
		onFromTimePickerFocus: PropTypes.func,
		onToDateChange: PropTypes.func,
		onToTimePickerBlur: PropTypes.func,
		onToTimePickerChange: PropTypes.func,
		onToTimePickerClick: PropTypes.func,
		onToTimePickerFocus: PropTypes.func,
		separatorDateTime: PropTypes.string,
		separatorTimeRange: PropTypes.string,
		shiftFocus: PropTypes.bool,
		toDate: PropTypes.instanceOf( Date ),
		toDateInput: PropTypes.string,
		toDateDisabled: PropTypes.bool,
		toDateFormat: PropTypes.string,
		toTime: PropTypes.string,
		toTimeDisabled: PropTypes.bool,
	};

	constructor( props ) {
		super( props );
		this.toDayPickerInput = createRef();
	}

	getFromDayPickerInputProps = () => {
		const {
			fromDate,
			fromDateInput,
			fromDateDisabled,
			fromDateFormat,
			onFromDateChange,
			shiftFocus,
			toDate,
		} = this.props;

		const props = {
			value: fromDateInput,
			format: fromDateFormat,
			formatDate: formatDate,
			parseDate: parseDate,
			dayPickerProps: {
				selectedDays: [ fromDate, { from: fromDate, to: toDate } ],
				disabledDays: { after: toDate },
				modifiers: {
					start: fromDate,
					end: toDate,
				},
				toMonth: toDate,
			},
			onDayChange: onFromDateChange,
			inputProps: {
				disabled: fromDateDisabled,
			},
		};

		/**
		 * If shiftFocus is true, selection of date on fromDayPickerInput
		 * automatically focuses on toDayPickerInput
		 */
		if ( shiftFocus ) {
			props.dayPickerProps.onDayClick = () => (
				this.toDayPickerInput.current.focus()
			);
		}

		return props;
	};

	getToDayPickerInputProps = () => {
		const {
			fromDate,
			onToDateChange,
			shiftFocus,
			toDate,
			toDateInput,
			toDateDisabled,
			toDateFormat,
		} = this.props;

		const props = {
			value: toDateInput,
			format: toDateFormat,
			formatDate: formatDate,
			parseDate: parseDate,
			dayPickerProps: {
				selectedDays: [ fromDate, { from: fromDate, to: toDate } ],
				disabledDays: { before: fromDate },
				modifiers: {
					start: fromDate,
					end: toDate,
				},
				month: fromDate,
				fromMonth: fromDate,
			},
			onDayChange: onToDateChange,
			inputProps: {
				disabled: toDateDisabled,
			},
		};

		/**
		 * If shiftFocus is true, selection of date on fromDayPickerInput
		 * automatically focuses on toDayPickerInput
		 */
		if ( shiftFocus ) {
			props.ref = this.toDayPickerInput;
		}

		return props;
	};

	getFromTimePickerProps = () => {
		const {
			fromTime,
			fromTimeDisabled,
			onFromTimePickerBlur,
			onFromTimePickerChange,
			onFromTimePickerClick,
			onFromTimePickerFocus,
		} = this.props;

		const props = {
			current: fromTime,
			start: time.START_OF_DAY,
			end: time.END_OF_DAY,
			onBlur: onFromTimePickerBlur,
			onChange: onFromTimePickerChange,
			onClick: onFromTimePickerClick,
			onFocus: onFromTimePickerFocus,
			timeFormat: date.FORMATS.WP.time,
			disabled: fromTimeDisabled,
		};

		return props;
	};

	getToTimePickerProps = () => {
		const {
			onToTimePickerBlur,
			onToTimePickerChange,
			onToTimePickerClick,
			onToTimePickerFocus,
			toTime,
			toTimeDisabled,
		} = this.props;

		const props = {
			current: toTime,
			start: time.START_OF_DAY,
			end: time.END_OF_DAY,
			onBlur: onToTimePickerBlur,
			onChange: onToTimePickerChange,
			onClick: onToTimePickerClick,
			onFocus: onToTimePickerFocus,
			timeFormat: date.FORMATS.WP.time,
			disabled: toTimeDisabled,
		};

		return props;
	};

	render() {
		const {
			className,
			separatorDateTime,
			separatorTimeRange,
		} = this.props;

		return (
			<div className={ classNames( 'tribe-editor__date-time-range-picker', className ) }>
				<div className="tribe-editor__date-time-range-picker__start">
					<DayPickerInput { ...this.getFromDayPickerInputProps() } />
					<span
						className={ classNames(
							'tribe-editor__date-time-range-picker__separator',
							'tribe-editor__date-time-range-picker__separator--date-time',
						) }
					>
						{ separatorDateTime }
					</span>
					<TimePicker { ...this.getFromTimePickerProps() } />
				</div>
				<div className="tribe-editor__date-time-range-picker__end">
					<span
						className={ classNames(
							'tribe-editor__date-time-range-picker__separator',
							'tribe-editor__date-time-range-picker__separator--time-range',
						) }
					>
						{ separatorTimeRange }
					</span>
					<DayPickerInput { ...this.getToDayPickerInputProps() } />
					<span
						className={ classNames(
							'tribe-editor__date-time-range-picker__separator',
							'tribe-editor__date-time-range-picker__separator--date-time',
						) }
					>
						{ separatorDateTime }
					</span>
					<TimePicker { ...this.getToTimePickerProps() } />
				</div>
			</div>
		);
	}
}

export default DateTimeRangePicker;
