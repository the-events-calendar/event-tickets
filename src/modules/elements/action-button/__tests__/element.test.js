/**
 * External dependencies
 */
import React from 'react';
import renderer from 'react-test-renderer';

/**
 * Internal dependencies
 */
import { ActionButton } from '@moderntribe/tickets/elements';
import { positions } from '@moderntribe/tickets/elements/action-button/element';
import { Button } from '@moderntribe/common/elements';

const Icon = () => ( <span role="img" aria-label="Emoji">ðŸ¦–</span> );

describe( 'ActionButton', () => {
	test( 'component rendered', () => {
		const component = renderer.create(
			<ActionButton icon={ <Icon /> }>Custom Action</ActionButton>,
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	test( 'component rendered with the correct class when icon is on the right', () => {
		const component = renderer.create(
			<ActionButton icon={ <Icon /> } position={ positions.right }>Custom Action</ActionButton>,
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	test( 'component positions', () => {
		expect( positions.right ).toBe( 'right' );
		expect( positions.left ).toBe( 'left' );
	} );

	test( 'component has class', () => {
		const component = mount( <ActionButton icon={ <Icon /> }>Custom Action</ActionButton> );
		const button = component.find( Button );
		expect( button.hasClass( 'tribe-editor__action-button' ) ).toBe( true );
	} );

	test( 'component rendered as link', () => {
		const component = renderer.create(
			<ActionButton asLink={ true } icon={ <Icon /> } href="#">Test Action</ActionButton>,
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );
} );
