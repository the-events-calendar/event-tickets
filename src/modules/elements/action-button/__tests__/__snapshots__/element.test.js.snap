// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton component rendered 1`] = `
<button
  className="tribe-editor__button tribe-editor__action-button tribe-editor__action-button--icon-left"
  onClick={[Function]}
  type="button"
>
  <span
    aria-label="Emoji"
    role="img"
  >
    ðŸ¦–
  </span>
  <span
    className="tribe-editor__action-button__label"
  >
    Custom Action
  </span>
</button>
`;

exports[`ActionButton component rendered as link 1`] = `
<a
  className="tribe-editor__link tribe-editor__action-button tribe-editor__action-button--icon-left"
  href="#"
  onMouseDown={[Function]}
>
  <span
    aria-label="Emoji"
    role="img"
  >
    ðŸ¦–
  </span>
  <span
    className="tribe-editor__action-button__label"
  >
    Test Action
  </span>
</a>
`;

exports[`ActionButton component rendered with the correct class when icon is on the right 1`] = `
<button
  className="tribe-editor__button tribe-editor__action-button tribe-editor__action-button--icon-left"
  onClick={[Function]}
  placement="right"
  type="button"
>
  <span
    aria-label="Emoji"
    role="img"
  >
    ðŸ¦–
  </span>
  <span
    className="tribe-editor__action-button__label"
  >
    Custom Action
  </span>
</button>
`;
